### PR TITLE
refactor: use decimal as min_base_fee

### PR DESCRIPTION
**Pull request by MSalopek**
This PR changes the feemarket implementation to use a decimal instead of an int for the `min_base_fee` field.

Not using a decimal could cause an inadvertant and unexpeced rise in the gas fee costs.

In some calculations I was not sure about the correctnes of the operation. Search for `// Note: not sure about` across the repo.

Feedback is appreciated.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -79,8 +79,8 @@ message State {
   // BaseFee is the current base fee. This is denominated in the fee per gas
   // unit.
   string base_fee = 1 [
-    (cosmos_proto.scalar) = "cosmos.Int",
-    (gogoproto.customtype) = "cosmossdk.io/math.Int",
+    (cosmos_proto.scalar) = "cosmos.Legacy",
```
Comment (aljo242) [inline README.md:82]:
```suggestion
    (cosmos_proto.scalar) = "cosmos.Dec",
```

File: proto/feemarket/feemarket/v1/genesis.proto [hunk 2]
```diff
@@ -25,8 +25,8 @@ message State {
   // BaseFee is the current base fee. This is denominated in the fee per gas
   // unit.
   string base_fee = 1 [
-    (cosmos_proto.scalar) = "cosmos.Int",
-    (gogoproto.customtype) = "cosmossdk.io/math.Int",
+    (cosmos_proto.scalar) = "cosmos.Legacy",
```
Comment (aljo242) [inline proto/feemarket/feemarket/v1/genesis.proto:28]:
```suggestion
    (cosmos_proto.scalar) = "cosmos.Dec",
```

**Review by aljo242 — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (MSalopek) [reply README.md:82]: Updated

File: proto/feemarket/feemarket/v1/genesis.proto (hunk 2 above)
Comment (MSalopek) [reply proto/feemarket/feemarket/v1/genesis.proto]: Updated

**Comment by aljo242** [comment]
@MSalopek I'm not so sure that moving to use a decimal value is going to be a good idea here.  Or at least this implementation of it.

In the event where the decimal value goes below 1, anytime the [TruncateInt](https://github.com/cosmos/cosmos-sdk/blob/80e1a45e706bb0db46813fdcc33f78eda66554a9/math/dec.go#L720) is called, the value is rounded _down_, meaning the resulting fee will be 0 right?

I don't thing there should ever be a case where the on-chain fee is 0.  Is this behavior you are intending?

**Comment by aljo242** [comment]
There should at least be logic that ensures that the resulting fee is non-zero

File: x/feemarket/post/fee.go [hunk 3]
```diff
@@ -72,8 +72,9 @@ func (dfd FeeMarketDeductDecorator) PostHandle(ctx sdk.Context, tx sdk.Tx, simul
 		return ctx, errorsmod.Wrapf(err, "unable to get fee market state")
 	}
 
-	baseFee := sdk.NewCoin(params.FeeDenom, state.BaseFee)
-	minGasPrices := sdk.NewCoins(baseFee)
+	// Note: not sure about the TruncateInt() method call
```
Comment (aljo242) [inline x/feemarket/post/fee.go:75]: remove

**Review by aljo242 — CHANGES_REQUESTED** [review]

File: x/feemarket/post/fee.go (hunk 3 above)
Comment (MSalopek) [reply x/feemarket/post/fee.go]: removed

**Review by aljo242 — APPROVED** [review]
Approving - but there are some caveats here:

This code must be tested much further before it is brought into production.  I would recommend expanding on the `tests/e2e` directory we have significantly before bringing to prod. Load tests as well

**Pull request merged**